api/client/proto/authservice.proto

```diff
@@ -747,9 +746,7 @@ enum DeviceType {
 // MFAAuthenticateChallenge is a challenge for all MFA devices registered for a
 // user.
 message MFAAuthenticateChallenge {
-    // U2F contains one U2FChallenge per U2F device registered for a
-    // user. Each challenge is unique.
-    repeated U2FChallenge U2F = 1;
+    reserved 1; // repeated U2FChallenge U2F
```

Comment:
just curious why this one has to be reserved?

Reply:
`reserved` is so the field tag can't be reused. I'm mostly playing it safe, but tag reuses can cause bugs that are difficult to reason about. If server and client drift in version they may think the same tag represents different fields/types, likely cause somewhat cryptic gRPC errors (or, in worst case scenario, bogus data).

Entire messages are safe to be deleted. For fields, I've reserved all I could, except for oneof fields, which can't be reserved.